### ISSUE TITLE
Crash in terraform v0.6.14 on win7 enterprise x64

### ISSUE THREAD

**Issue by l3m** [hashicorp/terraform#6280]
### Terraform Version
v0.6.14 

### Affected Resource(s)
Please list the resources as a list, for example:
- azurerm

### Terraform Configuration Files

The unmodified example file from https://www.terraform.io/docs/providers/azurerm/

https://gist.github.com/l3m/3908cb95f556c2155a0194de43707578

### Panic Output

https://gist.github.com/l3m/af0c4c0d45e6b447a7f22c48d79c6051

### Expected Behavior
No crash

### Actual Behavior
Crash

### Steps to Reproduce
terraform plan

### Important Factoids
- Install on windows with Chocolatey
- Run from within a git bash in ConEmu

**Comment by stack72** [comment]
Hi @l3m 

Apologies for the issue here. It seems like there is an issue with the credentials you are using:

```
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: panic: runtime error: invalid memory address or nil pointer dereference
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: [signal 0xc0000005 code=0x0 addr=0x10 pc=0x68916e]
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: 
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: goroutine 49 [running]:
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: panic(0xd49100, 0xc0820020a0)
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: 	/opt/go/src/runtime/panic.go:464 +0x3f4
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: github.com/hashicorp/terraform/vendor/github.com/Azure/azure-sdk-for-go/arm/resources/resources.ProvidersClient.Register(0x3f50148, 0xc08222c000, 0x3f50170, 0xc082204e00, 0x0, 0x0, 0xf1af30, 0x13, 0x0, 0xd18c2e2800, ...)
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: 	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/Azure/azure-sdk-for-go/arm/resources/resources/providers.go:206 +0x21e
2016/04/21 15:25:24 [DEBUG] terraform-provider-azurerm.exe: github.com/hashicorp/terraform/builtin/providers/azurerm.registerAzureResourceProvidersWithSubscription(0xc082218050, 0xc08222e000, 0x0, 0x0)
```

Can you tell me how you created these? One of the people who got bit by this in the past wrote the instructions on how they solved it https://www.terraform.io/docs/providers/azurerm/index.html

Paul

**Comment by l3m** [comment]
Yes, it's definitely an issue with the credentials - as I wrote, I just copied the default example file from the terraform website and ran terraform plan, just to see what happens. I didn't yet fill in the credentials (they were still set to '...'). 

I would assume that the correct behavior would be to print out some error message about invalid credentials or so, not to crash.

**Comment by stack72** [comment]
@l3m 

got it - yes, we are going to add a message there for sure :) It's something that slipped through the cracks 

P.

**Comment by stack72** [comment]
Fixed in #6290
